sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -334,6 +340,160 @@ private case class ParquetFileInfoWithBlockMeta(filePath: Path, blocks: Seq[Bloc
     partValues: InternalRow, schema: MessageType, isCorrectedInt96RebaseMode: Boolean,
     isCorrectedRebaseMode: Boolean, hasInt96Timestamps: Boolean)
 
+class HMBSeekableInputStream(hmb: HostMemoryBuffer, hmbLength: Long) extends SeekableInputStream {
```

Comment:
Seems like this belongs in `HostMemoryStreams.scala` and could reuse code from `HostMemoryInputStream` (or maybe `HostMemoryInputStream` should just add the `SeekableInputStream` functionality).

Reply:
The reason I did this was because SeekableInputStream is a Parquet specific class. I wasn't sure I should be mixing those two. If you still want me to I am happy to do it.